app/sidekiq/terms_of_use/sign_up_service_updater_job.rb

```diff
@@ -13,8 +13,23 @@ class SignUpServiceUpdaterJob
       attr_package_key = job['args'].first
       attrs = Sidekiq::AttrPackage.find(attr_package_key)
 
-      Rails.logger.warn('[TermsOfUse][SignUpServiceUpdaterJob] retries exhausted',
-                        { icn: attrs&.dig(:icn), exception_message: exception.message, attr_package_key: })
+      icn = attrs&.dig(:icn)
+      version = attrs&.dig(:version)
+      agreement = UserAccount.find_by(icn:)&.terms_of_use_agreements&.where(agreement_version: version)&.last
```

Comment:
You could refactor this to return the `TermsOfUseAgreement` without the safe navigation operators. I think this query would do the trick.

```ruby
TermsOfUseAgreement.joins(:user_account)
                   .where(user_accounts: { icn: })
                   .where(agreement_version: version)
                   .order(created_at: :desc)
                   .first
```